src/agents/OpenCodeAgent.ts

```diff
@@ -0,0 +1,87 @@
+import * as path from 'path';
+import { IAgent, IAgentConfig } from './IAgent';
+import { backupFile, writeGeneratedFile } from '../core/FileSystemUtils';
+
+/**
+ * OpenCode AI agent adapter.
+ */
+export class OpenCodeAgent implements IAgent {
+  getIdentifier(): string {
+    return 'opencode';
+  }
+
+  getName(): string {
+    return 'OpenCode';
+  }
+
+  async applyRulerConfig(
+    concatenatedRules: string,
+    projectRoot: string,
+    rulerMcpJson: Record<string, unknown> | null,
+    agentConfig?: IAgentConfig,
+  ): Promise<void> {
+    const output =
+      agentConfig?.outputPath ?? this.getDefaultOutputPath(projectRoot);
+    await backupFile(output);
+
+    // Prepend the OpenCode-specific header to the content
+    const header = `# opencode agent guidelines
```

Comment:
Why is this hardcoded? That's something that should live in .ruler/ as instructions.